src/Sarif.Converters/PREFastConverter.cs

```diff
@@ -112,16 +115,21 @@ private Result CreateResult(Defect defect)
             return result;
         }
 
-        private void AddLogicalLocation(string name, string decoratedName)
+        private int AddLogicalLocation(string name, string decoratedName)
         {
-            // Use the FQLN so the correct dictionary key is generated by AddLogicalLocation
+            string fullyQualifiedName = name;
+
+            name = name.Contains("::") ? name.Substring(name.LastIndexOf("::") + "::".Length) : null;
```

Comment:
>name [](start = 12, length = 4)

As I noted in a comment on one of the test files, this logic is actually backwards. `name` is required, and `fullyQualifiedName` is required unless it is the same as `name` (a top-level location).
```
string fullyQualifiedName = name;

int lastScopeOperatorPosition = name.Substring(name.LastIndexOf(ScopeOperator));
if (lastScopeOperatorPosition != -1)
{
    // This is a nested logical location, so "name" is the rightmost component.
    name = name.Substring(lastScopeOperatorPosition + ScopeOperator.Length)
}
else
{
    // This is a top-level logical location; fullyQualifiedName is the same as name, so
    // it's not necessary to emit it.
    fullyQualifiedName =null;
}
```

Reply:
Yes. [https://github.com/Microsoft/sarif-sdk/issues/1173](https://github.com/Microsoft/sarif-sdk/issues/1173) 

---
In reply to: [243865028](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243865028) [](ancestors = 243865028)